hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/EncryptedFileSystem.java

```diff
@@ -98,6 +98,15 @@ public boolean rename(Path src, Path dst) throws IOException {
         return renamed;
     }
 
+    @Override
+    public boolean delete(Path path, boolean recursive) throws IOException {
+        if (!fs.delete(path, recursive)) {
+            return false;
+        }
+        keyStore.remove(path.toString());
```

Comment:
If we want to handle retries here, in case the key store fails to remove the key material but the file was deleted, we'll have to do an existence check on the file first.